lib/net.js

```diff
@@ -1166,6 +1248,79 @@ function lookupAndConnect(self, options) {
   debug('connect: dns options', dnsopts);
   self._host = host;
   const lookup = options.lookup || dns.lookup;
+
+  if (dnsopts.family !== 4 && dnsopts.family !== 6 && autoDetectFamily) {
+    debug('connect: autodetecting family via happy eyeballs');
+
+    dnsopts.all = true;
+
+    defaultTriggerAsyncIdScope(self[async_id_symbol], function() {
+      lookup(host, dnsopts, function emitLookup(err, addresses) {
+        // It's possible we were destroyed while looking this up.
+        // XXX it would be great if we could cancel the promise returned by
```

Comment:
`XXX` should follow the same conventions as `TODO` and `FIXME`:

```suggestion
        // XXX(ShogunPanda): it would be great if we could cancel the promise returned by
```

Reply:
Actually I have no idea how this belongs to. I just a copy and paste of the existing method. I would leave as it is.